src/Streamly/Internal/Data/Fold/Types.hs

```diff
@@ -353,6 +354,16 @@ lmapM f (Fold step begin done) = Fold step' begin done
   where
     step' x a = f a >>= step x
 
+-- | @(mapMaybe f fold)@ maps a 'Maybe' returning function @f@ on the input of the fold.
+-- filter out the 'Nothing' elements, and return a values extracted from 'Just'
+-- >>> S.fold (FL.mapMaybe (\x -> case x of  5 -> Just x ; _ -> Nothing) FL.sum) 
+-- (S.enumerateFromTo 1 10) 
+-- 5
+-- @since 0.8.1
```

Comment:
We usually keep the combinators internal for a while and then expose them later once we know they are stable and should be exposed. So here we can mark it `Internal` instead of annotating with a release. The markup would be:

```
/Internal/
```

Also, let's make it `INLINE` instead of `INLINABLE`. We have marked some functions `INLINABLE` but we really want these `INLINE` for guaranteed fusion.

Reply:
Let's add a benchmark and test as well for this. See `benchmark/Streamly/Benchmark/Data/Fold.hs` for existing benchmarks and `test/Streamly/Test/Internal/Data/Fold.hs` for tests.

Reply:
In the docs, need a comma instead of period:

```
on the input of the fold, filter out the 'Nothing' elements,
```